src/certificate.js

```diff
@@ -94,10 +120,56 @@ function verifyCertificate(certificate) {
   return true;
 }
 
+Certificate.prototype.privacyFilter = function _privacyFilter(fields) {
+  this.certificate = _.cloneDeep(this.certificate);
+
+  const { evidence, privateEvidence } = this.certificate.badge;
+  const { type, saltLength } = this.certificate.badge.evidencePrivacyFilter;
+  if (!type) throw new Error("Privacy filter algorithm cannot be found");
+  if (!saltLength) throw new Error("Privacy salt length cannot be found");
+  if (type !== "SaltedProof") throw new Error("Unsupported privacy filter");
+
+  const valueToRemove = fields instanceof Array ? fields : [fields];
+
+  const flattenedEvidence = flatten(evidence);
```

Comment:
We can use lodash to improve the performance of this. (Rather than flattening and unflattening)

```javascript
// Pick out the evidence we want to privatise
const privateEvidences = _.pick(evidence, valuesToRemove);

// Unset the privatised evidence fields
valuesToRemove.forEach(path => {
  _.unset(evidence, path)
});
```